modules/core/src/main/scala/schema.scala

```diff
@@ -23,12 +23,57 @@ trait Schema {
 
   def pos: SourcePos
 
-  /** The types defined by this `Schema`. */
-  def types: List[NamedType]
+  /** The types defined by this `Schema` prior to any extensions. */
+  def unExtendedTypes: List[NamedType]
+
+  /** The types defined by this `Schema` with any extensions applied. */
+  def types: List[NamedType] = {
```

Comment:
With this implementation approach this has to be lazy rather than eager owing to the way schemas are constructed using mutable vars

Reply:
I think this will need to be a `lazy val` for performance reasons.